plugin.js

```diff
@@ -63,7 +63,7 @@ function onReqHandlerWrapper (fastify) {
 
 function plugin (fastify, options, next) {
   const secret = options.secret
-  const addHookOnRequest = options.addHookOnRequest ?? true
+  const hook = options.hook ?? 'onRequest'
```

Comment:
looks odd and could result in a error. I could set hook to true. and it result in an error. 

maybe write a little function which returns onRequest if it is true, false if false, string return the string.

Reply:
ok, may throw an error if the hook doesn't match 'preHandler', 'onRequest', 'preValidation' and false?

Reply:
Sure thing.